### PR TITLE
Panel class name dropped bug

### PR DESCRIPTION
**Pull request by ghsolomon**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by lbwexler** [comment]
cr: Tom

File: desktop/cmp/panel/Panel.ts [hunk 1]
```diff
@@ -196,7 +196,13 @@ export const [Panel, panel] = hoistCmp.withFactory<PanelProps>({
 
         // 3) Wrap in modal support if needed
         if (modalSupportModel) {
-            item = modalSupport({model: modalSupportModel, item});
+            item = modalSupport({
+                model: modalSupportModel,
+                item: frame({
+                    item,
+                    className: model.isModal ? className : undefined
+                })
```
Comment (amcclain) [inline desktop/cmp/panel/Panel.ts:204]: This is another layer of DOM nesting we're introducing here?  Looks like this ends up as an anonymous frame without any class or identifying info if we aren't specifying a classname? 

Sorry to be a pain here, it's just that we do sometimes need to target CSS selectors pretty specifically so (changes to / extra) DOM layers do end up mattering. I would like to understand this better and am not 100% on what we're doing here.

Comment (TomTirapani) [reply desktop/cmp/panel/Panel.ts:204]: I understand the concerns here (and had the same), but couldn't think of any other option that would avoid having to remount the content.

There are already several DOM layers between the outer wrapper ("xh-panel") and the inner content ("xh-panel__content"), the number and composition of which vary depending on which features you are using (e.g. resizable, modal support, etc). None of our styling assumes that `xh-panel__content` is a direct child of `xh-panel` for this reason.

Given the above, I figured it was unlikely to break styling in client apps and worthwhile compromise to support modal support properly.

Comment (amcclain) [reply desktop/cmp/panel/Panel.ts:204]: Thanks for the extra context - appreciate it, and just walked through with Greg also.  We added an additional line to the CL entry to call out the DOM change for the record.

Key takeaway being that the DOM is not really an API and that developers should not assume an overly specific hierarchy. Some seemingly "anonymous" elements might be present for internal implementation purposes that won't be immediately obvious when looking at the DOM itself.

**Pull request merged**